File: bus-micro-service/micro-service-i/src/main/java/com/shaun/microservice/microservicei/router/RestRouter.java
package com.shaun.microservice.microservicei.router;


import org.apache.camel.builder.RouteBuilder;
import org.springframework.stereotype.Component;

@Component
public class RestRouter extends RouteBuilder {

    @Override
    public void configure() throws Exception {
        from("timer:hello?period={{timer.period}}").routeId("hello").routeGroup("hello-group")
                .transform().method("myBean", "saySomething")
                .filter(simple("${body} contains 'foo'"))
                .to("log:foo")
                .end()
                .to("stream:out");
    }

}

File: bus-micro-service/micro-service-i/src/main/kotlin/com/shaun/microservice/microservicei/camelrouter/RestRouter.kt
package com.shaun.microservice.microservicei.camelrouter

import org.apache.camel.builder.RouteBuilder
import org.springframework.stereotype.Component

@Component
class RestRouter : RouteBuilder() {
    override fun configure() {
        from("timer://system?period={{timer.period}}").routeId("hello").routeGroup("hello-group")
                .transform().method("myBean", "saySomething")
                .filter(simple("\${body} contains 'foo'"))
                .to("log:foo")
                .end()
                .to("stream:out")
    }
}